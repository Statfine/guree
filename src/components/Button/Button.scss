@import "../../common/utils/index.scss";

.#{scopedClass(button)} {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid currentColor;
  border-radius: 0.3em;
  cursor: pointer;

  & + & {
    margin-left: 8px;
  }
  &.large {
    padding: 10px 30px;
  }
  &.middle {
    padding: 6px 20px;
  }
  &.small {
    padding: 2px 10px;
  }
  > .action + span {
    margin-left: 6px;
  }
  &.default {
    color: #666;
    background-color: #fff;
  }
  &.primary {
    color: #fff;
    background-color: #409eff;
  }
  &.danger {
    color: #fff;
    background-color: #f56c6c;
  }
  &.warning {
    color: #fff;
    background-color: #e6a23c;
  }
  &.text {
    color: #fff;
    background-color: #67c23a;
  }
  &[disabled] {
    opacity: 0.35;
    cursor: default;
  }

  &:focus{
    outline: none;
    opacity: 0.8;
  }
}
