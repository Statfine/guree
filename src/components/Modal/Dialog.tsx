import React, { ReactNode, Fragment } from "react";
import ReactDOM from "react-dom";
import "./Modal.scss";
import { combineClasses, scopedClass, classesObj } from "@utils/index";

interface Props {
  children?: ReactNode;
  mask?: boolean;
  onClickMask?: React.MouseEventHandler<HTMLElement>;
  visible?: boolean;
}

const Dialog: React.FunctionComponent<Props> = ({
  children,
  mask,
  onClickMask,
  visible,
}: Props) => {
  const modalClass = (...classes: (string | Array<string> | classesObj)[]) =>
    scopedClass("modal", ...classes);

  const maskClassNames = combineClasses(modalClass("mask"), {
    [`${modalClass("hidden")}`]: !visible,
  });

  const modalClassNames = combineClasses(modalClass(), {
    [`${modalClass("hidden")}`]: !visible,
  });

  return ReactDOM.createPortal(
    <Fragment>
      {mask ? (
        <div
          className={maskClassNames}
          onClick={onClickMask}
        ></div>
      ) : null}
      <div className={modalClassNames}>{children}</div>
    </Fragment>,
    document.body
  );
};

export default Dialog;
