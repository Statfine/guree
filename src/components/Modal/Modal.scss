@import "../../common/utils/index.scss";

@function modal-class($cls) {
  @return scopedClass();
}

.#{scopedClass(modal-mask)} {
  background-color: rgba($color: #000000, $alpha: 0.5);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1998;
}

.#{scopedClass(modal)} {
  border: 1px rgba($color: #000000, $alpha: 0.5) solid;
  background-color: #ffffff;
  position: fixed;
  z-index: 1999;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  min-width: 420px;
  border-radius: 5px;
  &-hidden {
    display: none;
  }
  &-close {
    width: 40px;
    height: 40px;
    position: absolute;
    top: 0;
    right: 0;
    @include flex-center-center();
    cursor: pointer;
  }
  &-header {
    min-height: 40px;
    padding: 10px 20px 10px;
    .#{scopedClass(modal-title)} {
      @include flex-start-center();
      min-height: 20px;
      padding-right: 20px;
    }
  }
  &-body {
    padding: 20px 20px;
  }
  &-footer {
    padding: 10px 20px 20px;
    min-height: 60px;
    @include flex-end-center();
  }
  &-confirm {
    &-body {
      font-size: 22px;
      margin-right: 10px;
    }
  }
}
