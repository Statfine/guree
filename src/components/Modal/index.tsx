export { default } from "./Modal";
export { $modal, $alert, $confirm } from "./renderModal";
