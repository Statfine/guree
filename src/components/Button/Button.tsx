import React from "react";

import "./Button.scss";

/**
 * The only true button.
 */

export interface Props {
  children: string;
}

export default function Button({ children }: Props) {
  return <button className="button">{children}</button>;
}
