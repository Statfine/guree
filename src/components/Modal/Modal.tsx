import React, { ReactNode, Fragment } from "react";
import "./Modal.scss";
import Dialog from "./Dialog";
import { scopedClass, classesObj } from "@utils/index";
import Icon from "@components/Icon";
import Button from "@components/Button";

export interface Props {
  children?: ReactNode;
  /** 是否存在遮罩 */
  mask?: boolean;
  /** 点击遮罩时是否关闭 */
  closeOnClickMask?: boolean;
  /** 标签内容 */
  title?: ReactNode;
  /** 点击确定时调用 */
  onConfirm?: React.MouseEventHandler<HTMLElement>;
  /** 点击取消，右上角关闭按钮，遮罩时调用 */
  onCancel?: React.MouseEventHandler<HTMLElement>;
  /** 是否可见 */
  visible?: boolean;
  /** 底部内容 */
  footer?: ReactNode;
  /** 是否显示右上角关闭按钮 */
  closeable?: boolean;
  /** 取消按钮内容 */
  cancelContent?: ReactNode;
  /** 确定按钮内容 */
  confirmContent?: ReactNode;
}

const Modal: React.FunctionComponent<Props> = ({
  children,
  mask = true,
  closeOnClickMask = true,
  title,
  onConfirm,
  onCancel,
  visible = false,
  footer,
  closeable = true,
  cancelContent = "取消",
  confirmContent = "确定",
}: Props) => {
  const modalClass = (...classes: (string | Array<string> | classesObj)[]) =>
    scopedClass("modal", ...classes);

  const clickMaskHandler: React.MouseEventHandler<HTMLElement> = (e) => {
    closeOnClickMask && cancelHandler(e);
  };

  const cancelHandler: React.MouseEventHandler<HTMLElement> = (e) => {
    onCancel && onCancel(e);
  };

  const confirmHandler: React.MouseEventHandler<HTMLElement> = (e) => {
    onConfirm && onConfirm(e);
  };

  const footerNodes =
    typeof footer === "undefined" ? (
      <Fragment>
        <Button onClick={cancelHandler}>{cancelContent}</Button>
        <Button onClick={confirmHandler} type="primary">
          {confirmContent}
        </Button>
      </Fragment>
    ) : (
      footer
    );

  return (
    <Dialog visible={visible} mask={mask} onClickMask={clickMaskHandler}>
      {closeable ? (
        <div className={modalClass("close")} onClick={cancelHandler}>
          <Icon name="times"></Icon>
        </div>
      ) : null}
      <div className={modalClass("header")}>
        <div className={modalClass("title")}>{title}</div>
      </div>
      <div className={modalClass("body")}>{children}</div>
      <div className={modalClass("footer")}>{footerNodes}</div>
    </Dialog>
  );
};

export default Modal;
