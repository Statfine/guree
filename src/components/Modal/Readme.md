### 基本用法

```jsx
import Button from "@components/Button";
import { useState } from "react";

const [visible, setVisible] = useState(false);
const showModal = () => {
  setVisible(true);
};

const hideModal = () => {
  setVisible(false);
};

const cancelHandler = (e) => {
  console.log(e);
  hideModal();
};

const confirmHandler = (e) => {
  console.log(e);
  hideModal();
};

<div>
  <Button onClick={showModal}>show</Button>

  <Modal
    title="title"
    visible={visible}
    onCancel={cancelHandler}
    onConfirm={confirmHandler}
  >
    <div>some things</div>
    <div>some things</div>
    <div>some things</div>
  </Modal>
</div>;
```

### 不显示遮罩

```jsx
import Button from "@components/Button";
import { useState } from "react";

const [visible, setVisible] = useState(false);
const showModal = () => {
  setVisible(true);
};

const hideModal = () => {
  setVisible(false);
};

const cancelHandler = (e) => {
  console.log(e);
  hideModal();
};

const confirmHandler = (e) => {
  console.log(e);
  hideModal();
};

<div>
  <Button onClick={showModal}>show</Button>

  <Modal
    title="title"
    visible={visible}
    onCancel={cancelHandler}
    onConfirm={confirmHandler}
    mask={false}
  >
    <div>some things</div>
    <div>some things</div>
    <div>some things</div>
  </Modal>
</div>;
```

### 自定义按钮文字

```jsx
import Button from "@components/Button";
import { useState } from "react";

const [visible, setVisible] = useState(false);
const showModal = () => {
  setVisible(true);
};

const hideModal = () => {
  setVisible(false);
};

const cancelHandler = (e) => {
  console.log(e);
  hideModal();
};

const confirmHandler = (e) => {
  console.log(e);
  hideModal();
};

<div>
  <Button onClick={showModal}>show</Button>

  <Modal
    title="title"
    visible={visible}
    onCancel={cancelHandler}
    onConfirm={confirmHandler}
    cancelContent="cancel"
    confirmContent="confirm"
  >
    <div>some things</div>
    <div>some things</div>
    <div>some things</div>
  </Modal>
</div>;
```

### 自定义底部内容

```jsx
import Button from "@components/Button";
import { useState } from "react";

const [visible, setVisible] = useState(false);
const showModal = () => {
  setVisible(true);
};

const hideModal = () => {
  setVisible(false);
};

const cancelHandler = (e) => {
  console.log(e);
  hideModal();
};

const confirmHandler = (e) => {
  console.log(e);
  hideModal();
};

const footer = [
  <Button key="no" onClick={cancelHandler} type="primary">
    NO
  </Button>,
  <Button key="yes" onClick={confirmHandler} type="danger">
    Yes
  </Button>,
];

<div>
  <Button onClick={showModal}>show</Button>

  <Modal
    title="warnning"
    visible={visible}
    onCancel={cancelHandler}
    onConfirm={confirmHandler}
    footer={footer}
  >
    <p>Are you sure ?</p>
  </Modal>
</div>;
```

### 不显示右上角关闭按钮

```jsx
import Button from "@components/Button";
import { useState } from "react";

const [visible, setVisible] = useState(false);
const showModal = () => {
  setVisible(true);
};

const hideModal = () => {
  setVisible(false);
};

const cancelHandler = (e) => {
  console.log(e);
  hideModal();
};

const confirmHandler = (e) => {
  console.log(e);
  hideModal();
};

<div>
  <Button onClick={showModal}>show</Button>

  <Modal
    title="title"
    visible={visible}
    onCancel={cancelHandler}
    onConfirm={confirmHandler}
    closeable={false}
  >
    <div>some things</div>
    <div>some things</div>
    <div>some things</div>
  </Modal>
</div>;
```

### 使用特定的 Modal

如果不想使用 `<Modal></Modal>` 形式，也可以使用 `$modal()` 的形式使用。注意：当使用 `$modal` 时，不支持 `visible` 参数，可用返回的 `close` 方法隐藏 Modal
另外，还有 `$alert` `$confirm` 两个特定的 Modal 可以使用

```jsx
import Button from "@components/Button";
import { $modal, $alert, $confirm } from "@components/Modal";

const cancelHandler = (e) => {
  console.log(e);
};

const confirmHandler = (e) => {
  console.log(e);
};

const showModal = () => {
  const modal = $modal({
    title: "title",
    onCancel: cancelHandler,
    onConfirm: confirmHandler,
    children: (
      <div>
        modal content
        <Button
          onClick={() => {
            modal.close();
          }}
        >
          close me
        </Button>
      </div>
    ),
  });
};

const showAlert = () => {
  $alert({
    title: "alert title",
    content: "alert content",
  });
};

const showConfirm = () => {
  $confirm({
    title: "confirm title",
    content: "confirm content",
  });
};

<div>
  <Button onClick={showModal}>show modal</Button>
  <Button onClick={showAlert}>show alert</Button>
  <Button onClick={showConfirm}>show confirm</Button>
</div>;
```
