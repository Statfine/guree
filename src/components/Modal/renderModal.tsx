import React, { ReactNode } from "react";
import ReactDOM from "react-dom";
import "./Modal.scss";
import Modal, { Props } from "./Modal";
import Button from "@components/Button";

const $modal = ({
  children,
  mask = true,
  closeOnClickMask,
  title,
  onConfirm,
  onCancel,
  footer,
  closeable,
  cancelContent,
  confirmContent,
}: Props) => {
  const visible = true;
  const container = document.createDocumentFragment();

  const confirmHandler: React.MouseEventHandler<HTMLButtonElement> = (e) => {
    close();
    onConfirm && onConfirm(e);
  };

  const cancelHandler: React.MouseEventHandler<HTMLButtonElement> = (e) => {
    close();
    onCancel && onCancel(e);
  };

  const close = () => {
    ReactDOM.unmountComponentAtNode(container);
  };

  document.body.appendChild(container);
  const modal = (
    <Modal
      children={children}
      mask={mask}
      closeOnClickMask={closeOnClickMask}
      title={title}
      onConfirm={confirmHandler}
      onCancel={cancelHandler}
      visible={visible}
      footer={footer}
      closeable={closeable}
      cancelContent={cancelContent}
      confirmContent={confirmContent}
    ></Modal>
  );
  ReactDOM.render(modal, container);

  return {
    close,
    modal,
  };
};

interface ConfirmProps {
  content?: ReactNode;
  closeOnClickMask?: boolean;
  title?: ReactNode;
  onConfirm?: React.MouseEventHandler<HTMLButtonElement>;
  onCancel?: React.MouseEventHandler<HTMLButtonElement>;
  confirmContent?: ReactNode;
  cancelContent?: ReactNode;
}

const $alert = ({
  content,
  closeOnClickMask = true,
  title,
  onConfirm,
  onCancel,
  confirmContent = "确定",
}: ConfirmProps) => {
  const confirmHandler: React.MouseEventHandler<HTMLButtonElement> = (e) => {
    modal.close();
    onConfirm && onConfirm(e);
  };
  const footer = (
    <Button onClick={confirmHandler} type="primary">
      {confirmContent}
    </Button>
  );

  const modal = $modal({
    children: content,
    closeOnClickMask,
    title,
    onConfirm: onConfirm,
    onCancel: onCancel,
    footer,
  });

  return {
    modal: modal.modal
  }
};

const $confirm = ({
  content,
  closeOnClickMask = true,
  title,
  onConfirm,
  onCancel,
  cancelContent,
  confirmContent,
}: ConfirmProps) => {
  const modal =  $modal({
    children: content,
    closeOnClickMask,
    title,
    onConfirm: onConfirm,
    onCancel: onCancel,
    cancelContent,
    confirmContent,
  });

  return {
    modal: modal.modal
  }
};

export { $modal, $alert, $confirm };
