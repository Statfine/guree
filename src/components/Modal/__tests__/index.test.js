import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Modal, { $modal, $alert, $confirm } from "..";
import { mount } from "enzyme";
import Button from "@components/Button";

jest.mock("react-dom", () => ({
  createPortal: (node) => node,
  render: jest.fn(),
  unmountComponentAtNode: jest.fn(),
}));

describe("Modal 组件", () => {
  it("正确导出 Modal 组件", () => {
    const modal = renderer.create(<Modal></Modal>).toJSON();
    expect(modal).toMatchSnapshot();
  });

  it("测试 Modal 组件 mask 参数", () => {
    const modal = renderer.create(<Modal mask={false}></Modal>).toJSON();
    expect(modal).toMatchSnapshot();
  });

  it("测试 Modal 组件 closeOnClickMask 参数", () => {
    const fn = jest.fn();
    const modal = mount(<Modal closeOnClickMask={true} onCancel={fn}></Modal>);
    modal.find(".guree-modal-mask").simulate("click");
    expect(fn).toBeCalled();
  });

  it("测试 Modal 组件 title 参数", () => {
    const modal = renderer.create(<Modal title="title"></Modal>).toJSON();
    expect(modal).toMatchSnapshot();
  });

  it("测试 Modal 组件 onConfirm 参数", () => {
    const fn = jest.fn();
    const modal = mount(<Modal visible={true} onConfirm={fn}></Modal>);
    modal.find(".guree-button-primary").simulate("click");
    expect(fn).toBeCalled();
  });

  it("测试 Modal 组件 onCancel 参数", () => {
    const fn = jest.fn();
    const modal = mount(<Modal visible={true} onCancel={fn}></Modal>);
    modal.find(".guree-button-default").simulate("click");
    expect(fn).toBeCalled();
  });

  it("测试 Modal 组件 visible 参数", () => {
    const modal = renderer.create(<Modal visible={true}></Modal>).toJSON();
    expect(modal).toMatchSnapshot();
  });

  it("测试 Modal 组件 footer 参数", () => {
    const footer = [
      <Button key="no" type="primary">
        NO
      </Button>,
      <Button key="yes" type="danger">
        Yes
      </Button>,
    ];
    const modal = renderer.create(<Modal footer={footer}></Modal>).toJSON();
    expect(modal).toMatchSnapshot();
  });

  it("测试 Modal 组件 closeable 参数", () => {
    const modal = renderer.create(<Modal closeable={false}></Modal>).toJSON();
    expect(modal).toMatchSnapshot();
  });

  it("测试 Modal 组件 cancelContent 参数", () => {
    const modal = renderer
      .create(<Modal cancelContent="cancel content"></Modal>)
      .toJSON();
    expect(modal).toMatchSnapshot();
  });

  it("测试 Modal 组件 confirmContent 参数", () => {
    const modal = renderer
      .create(<Modal confirmContent="confirm content"></Modal>)
      .toJSON();
    expect(modal).toMatchSnapshot();
  });
});

describe("Modal 组件的高阶组件", () => {
  it("测试 $modal 组件", () => {
    const modal = $modal({
      title: "title",
      children: "modal content",
    });
    const Modal = renderer.create(modal.modal).toJSON();
    expect(Modal).toMatchSnapshot();
  });

  it("测试 $modal 组件 onConfirm 参数", () => {
    const fn = jest.fn();
    const modal = $modal({
      title: "title",
      children: "modal content",
      onConfirm: fn,
    });
    const Modal = mount(modal.modal);
    Modal.find(".guree-button-primary").simulate("click");
    expect(fn).toBeCalled();
  });

  it("测试 $modal 组件 onCancel 参数", () => {
    const fn = jest.fn();
    const modal = $modal({
      title: "title",
      children: "modal content",
      onCancel: fn,
    });
    const Modal = mount(modal.modal);
    Modal.find(".guree-button-default").simulate("click");
    expect(fn).toBeCalled();
  });

  it("测试 $alert 组件", () => {
    const alert = $alert({
      title: "alert title",
      content: "alert content",
    });
    const Alert = renderer.create(alert.modal).toJSON();
    expect(Alert).toMatchSnapshot();
  });

  it("测试 $alert 组件 onConfirm 参数", () => {
    const fn = jest.fn();
    const alert = $alert({
      title: "alert title",
      children: "alert content",
      onConfirm: fn,
    });
    const Alert = mount(alert.modal);
    Alert.find(".guree-button-primary").simulate("click");
    expect(fn).toBeCalled();
  });

  it("测试 $confirm 组件", () => {
    const confirm = $confirm({
      title: "confirm title",
      content: "confirm content",
    });
    const Confirm = renderer.create(confirm.modal).toJSON();
    expect(Confirm).toMatchSnapshot();
  });
});
