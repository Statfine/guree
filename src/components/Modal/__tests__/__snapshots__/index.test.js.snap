// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal 组件 正确导出 Modal 组件 1`] = `
Array [
  <div
    className="guree-modal-mask guree-modal-hidden"
    onClick={[Function]}
  />,
  <div
    className="guree-modal guree-modal-hidden"
  >
    <div
      className="guree-modal-close"
      onClick={[Function]}
    >
      <svg
        className="guree-icon"
        style={
          Object {
            "fill": undefined,
            "height": "1em",
            "width": "1em",
          }
        }
      >
        <use
          xlinkHref="#times"
        />
      </svg>
    </div>
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      />
    </div>
    <div
      className="guree-modal-body"
    />
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-default"
        onClick={[Function]}
      >
         
        <span>
          取消
        </span>
      </button>
      <button
        className="guree-button guree-button-middle guree-button-primary"
        onClick={[Function]}
      >
         
        <span>
          确定
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Modal 组件 测试 Modal 组件 cancelContent 参数 1`] = `
Array [
  <div
    className="guree-modal-mask guree-modal-hidden"
    onClick={[Function]}
  />,
  <div
    className="guree-modal guree-modal-hidden"
  >
    <div
      className="guree-modal-close"
      onClick={[Function]}
    >
      <svg
        className="guree-icon"
        style={
          Object {
            "fill": undefined,
            "height": "1em",
            "width": "1em",
          }
        }
      >
        <use
          xlinkHref="#times"
        />
      </svg>
    </div>
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      />
    </div>
    <div
      className="guree-modal-body"
    />
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-default"
        onClick={[Function]}
      >
         
        <span>
          cancel content
        </span>
      </button>
      <button
        className="guree-button guree-button-middle guree-button-primary"
        onClick={[Function]}
      >
         
        <span>
          确定
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Modal 组件 测试 Modal 组件 closeable 参数 1`] = `
Array [
  <div
    className="guree-modal-mask guree-modal-hidden"
    onClick={[Function]}
  />,
  <div
    className="guree-modal guree-modal-hidden"
  >
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      />
    </div>
    <div
      className="guree-modal-body"
    />
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-default"
        onClick={[Function]}
      >
         
        <span>
          取消
        </span>
      </button>
      <button
        className="guree-button guree-button-middle guree-button-primary"
        onClick={[Function]}
      >
         
        <span>
          确定
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Modal 组件 测试 Modal 组件 confirmContent 参数 1`] = `
Array [
  <div
    className="guree-modal-mask guree-modal-hidden"
    onClick={[Function]}
  />,
  <div
    className="guree-modal guree-modal-hidden"
  >
    <div
      className="guree-modal-close"
      onClick={[Function]}
    >
      <svg
        className="guree-icon"
        style={
          Object {
            "fill": undefined,
            "height": "1em",
            "width": "1em",
          }
        }
      >
        <use
          xlinkHref="#times"
        />
      </svg>
    </div>
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      />
    </div>
    <div
      className="guree-modal-body"
    />
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-default"
        onClick={[Function]}
      >
         
        <span>
          取消
        </span>
      </button>
      <button
        className="guree-button guree-button-middle guree-button-primary"
        onClick={[Function]}
      >
         
        <span>
          confirm content
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Modal 组件 测试 Modal 组件 footer 参数 1`] = `
Array [
  <div
    className="guree-modal-mask guree-modal-hidden"
    onClick={[Function]}
  />,
  <div
    className="guree-modal guree-modal-hidden"
  >
    <div
      className="guree-modal-close"
      onClick={[Function]}
    >
      <svg
        className="guree-icon"
        style={
          Object {
            "fill": undefined,
            "height": "1em",
            "width": "1em",
          }
        }
      >
        <use
          xlinkHref="#times"
        />
      </svg>
    </div>
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      />
    </div>
    <div
      className="guree-modal-body"
    />
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-primary"
      >
         
        <span>
          NO
        </span>
      </button>
      <button
        className="guree-button guree-button-middle guree-button-danger"
      >
         
        <span>
          Yes
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Modal 组件 测试 Modal 组件 mask 参数 1`] = `
<div
  className="guree-modal guree-modal-hidden"
>
  <div
    className="guree-modal-close"
    onClick={[Function]}
  >
    <svg
      className="guree-icon"
      style={
        Object {
          "fill": undefined,
          "height": "1em",
          "width": "1em",
        }
      }
    >
      <use
        xlinkHref="#times"
      />
    </svg>
  </div>
  <div
    className="guree-modal-header"
  >
    <div
      className="guree-modal-title"
    />
  </div>
  <div
    className="guree-modal-body"
  />
  <div
    className="guree-modal-footer"
  >
    <button
      className="guree-button guree-button-middle guree-button-default"
      onClick={[Function]}
    >
       
      <span>
        取消
      </span>
    </button>
    <button
      className="guree-button guree-button-middle guree-button-primary"
      onClick={[Function]}
    >
       
      <span>
        确定
      </span>
    </button>
  </div>
</div>
`;

exports[`Modal 组件 测试 Modal 组件 title 参数 1`] = `
Array [
  <div
    className="guree-modal-mask guree-modal-hidden"
    onClick={[Function]}
  />,
  <div
    className="guree-modal guree-modal-hidden"
  >
    <div
      className="guree-modal-close"
      onClick={[Function]}
    >
      <svg
        className="guree-icon"
        style={
          Object {
            "fill": undefined,
            "height": "1em",
            "width": "1em",
          }
        }
      >
        <use
          xlinkHref="#times"
        />
      </svg>
    </div>
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      >
        title
      </div>
    </div>
    <div
      className="guree-modal-body"
    />
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-default"
        onClick={[Function]}
      >
         
        <span>
          取消
        </span>
      </button>
      <button
        className="guree-button guree-button-middle guree-button-primary"
        onClick={[Function]}
      >
         
        <span>
          确定
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Modal 组件 测试 Modal 组件 visible 参数 1`] = `
Array [
  <div
    className="guree-modal-mask"
    onClick={[Function]}
  />,
  <div
    className="guree-modal"
  >
    <div
      className="guree-modal-close"
      onClick={[Function]}
    >
      <svg
        className="guree-icon"
        style={
          Object {
            "fill": undefined,
            "height": "1em",
            "width": "1em",
          }
        }
      >
        <use
          xlinkHref="#times"
        />
      </svg>
    </div>
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      />
    </div>
    <div
      className="guree-modal-body"
    />
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-default"
        onClick={[Function]}
      >
         
        <span>
          取消
        </span>
      </button>
      <button
        className="guree-button guree-button-middle guree-button-primary"
        onClick={[Function]}
      >
         
        <span>
          确定
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Modal 组件的高阶组件 测试 $alert 组件 1`] = `
Array [
  <div
    className="guree-modal-mask"
    onClick={[Function]}
  />,
  <div
    className="guree-modal"
  >
    <div
      className="guree-modal-close"
      onClick={[Function]}
    >
      <svg
        className="guree-icon"
        style={
          Object {
            "fill": undefined,
            "height": "1em",
            "width": "1em",
          }
        }
      >
        <use
          xlinkHref="#times"
        />
      </svg>
    </div>
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      >
        alert title
      </div>
    </div>
    <div
      className="guree-modal-body"
    >
      alert content
    </div>
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-primary"
        onClick={[Function]}
      >
         
        <span>
          确定
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Modal 组件的高阶组件 测试 $confirm 组件 1`] = `
Array [
  <div
    className="guree-modal-mask"
    onClick={[Function]}
  />,
  <div
    className="guree-modal"
  >
    <div
      className="guree-modal-close"
      onClick={[Function]}
    >
      <svg
        className="guree-icon"
        style={
          Object {
            "fill": undefined,
            "height": "1em",
            "width": "1em",
          }
        }
      >
        <use
          xlinkHref="#times"
        />
      </svg>
    </div>
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      >
        confirm title
      </div>
    </div>
    <div
      className="guree-modal-body"
    >
      confirm content
    </div>
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-default"
        onClick={[Function]}
      >
         
        <span>
          取消
        </span>
      </button>
      <button
        className="guree-button guree-button-middle guree-button-primary"
        onClick={[Function]}
      >
         
        <span>
          确定
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Modal 组件的高阶组件 测试 $modal 组件 1`] = `
Array [
  <div
    className="guree-modal-mask"
    onClick={[Function]}
  />,
  <div
    className="guree-modal"
  >
    <div
      className="guree-modal-close"
      onClick={[Function]}
    >
      <svg
        className="guree-icon"
        style={
          Object {
            "fill": undefined,
            "height": "1em",
            "width": "1em",
          }
        }
      >
        <use
          xlinkHref="#times"
        />
      </svg>
    </div>
    <div
      className="guree-modal-header"
    >
      <div
        className="guree-modal-title"
      >
        title
      </div>
    </div>
    <div
      className="guree-modal-body"
    >
      modal content
    </div>
    <div
      className="guree-modal-footer"
    >
      <button
        className="guree-button guree-button-middle guree-button-default"
        onClick={[Function]}
      >
         
        <span>
          取消
        </span>
      </button>
      <button
        className="guree-button guree-button-middle guree-button-primary"
        onClick={[Function]}
      >
         
        <span>
          确定
        </span>
      </button>
    </div>
  </div>,
]
`;
